src/components/surveys/survey-design/question-edit/SelectPhoneBottomMenu.tsx

```diff
@@ -286,18 +294,12 @@ const SelectPhoneBottomMenu: FunctionComponent<{
           {getOptions().map(optionKey => (
             <StyledMenuItem
               key={optionKey}
-              disabled={isDisabled(optionKey)}
+              disabled={isOptionAlreadyAdded(optionKey)}
               onClick={() => addGenericResponse(optionKey)}>
               {OPTIONS.get(optionKey)?.selectDisplayLabel}
             </StyledMenuItem>
           ))}
-          {step.baseType === 'string' ? (
-            <StyledMenuItem key="OTHER" disabled={isDisabled('OTHER')} onClick={() => addGenericResponse('OTHER')}>
-              + Add "Other"
-            </StyledMenuItem>
-          ) : (
-            <DisabledOtherMenuItem />
-          )}
+          {getOptions().length == 0 ? (<NoMenuItemOptions />) : (<></>) }
```

Comment:
@hallieswan Is there a way to do this where the UI component takes the filtered available options as an input and returns:

```
if (options.length == 0) {
<DisabledOtherMenuItem />
} else {
options.map(optionKey => (
            <StyledMenuItem
              key={optionKey}
              disabled={isOptionAlreadyAdded(optionKey)}
              onClick={() => addGenericResponse(optionKey)}>
              {OPTIONS.get(optionKey)?.selectDisplayLabel}
            </StyledMenuItem>
          ))
}
```

Or something? I spent way too much time trying to figure that out before giving up. 

Thanks!

Reply:
I think that wrapping the else logic in a [Fragment](https://react.dev/reference/react/Fragment) would work, which would look something like this: 

```TypeScript
{getOptions().length == 0 ? (
  <NoMenuItemOptions />
) : (
    <>
       {getOptions().map(optionKey => (
            <StyledMenuItem
              key={optionKey}
              disabled={isOptionAlreadyAdded(optionKey)}
              onClick={() => addGenericResponse(optionKey)}>
              {OPTIONS.get(optionKey)?.selectDisplayLabel}
            </StyledMenuItem>
          ))}
    </>
)}
```

Side note, this is [a common shortcut](https://react.dev/learn/conditional-rendering#logical-and-operator-) for rendering a component when a condition is true, but nothing otherwise: 

```TypeScript
{getOptions().length == 0 && <NoMenuItemOptions /> }
```

Reply:
Changed to use a function